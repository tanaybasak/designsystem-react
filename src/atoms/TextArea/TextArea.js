import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import prefix from '../../settings';

const TextArea = ({ className, ...restProps }) => {
    const [value, setValue] = useState(restProps.value || "");
    const classnames = `${prefix}-textarea ${className}`.trim();

    useEffect(() => {
        if (restProps.value || restProps.value === '') {
            setValue(restProps.value);
        }
    }, [restProps.value]);

    return (
        <textarea
            className={classnames}
            {...restProps}
            value={value}
            onChange={event => {
                setValue(event.currentTarget.value);
                restProps.onChange(event);
            }}
        />
    );
}

TextArea.propTypes = {
    /** Additional class name to be given to <input> tag. */
    className: PropTypes.string,
    /** The default Value of the field on rendering. */
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Specifying the <input> tag is disabled or not. */
    disabled: PropTypes.bool,
    /** Unique Id for the <input> tag. */
    id: PropTypes.string,
    /** Event to subscribe when the value of the Input field changes. */
    onChange: PropTypes.func,
    /** Event to subscribe when the Input field is clicked. */
    onClick: PropTypes.func,
    /** Event to subscribe when the Input field is focused. */
    onFocus: PropTypes.func,
    /** Event to subscribe when the Input field is Blurred. */
    onBlur: PropTypes.func,
    /** Placeholder text for the <input> tag. */
    placeholder: PropTypes.string,
    /** Value of the input field. */
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

TextArea.defaultProps = {
    className: "",
    disabled: false,
    onChange: () => { },
    onClick: () => { },
    onFocus: () => { },
    onBlur: () => { }
};

export default TextArea;