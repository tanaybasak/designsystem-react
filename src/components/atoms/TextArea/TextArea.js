import React, { useState } from './node_modules/react';
import PropTypes from './node_modules/prop-types';
import { prefix } from '../../../settings';

export default function TextArea({ className, ...restProps }) {
    const [value, setValue] = useState(restProps.value || '');
    const classnames = `${prefix}-textarea ${className}`.trim();

    return (
        <textarea
            className={classnames}
            {...restProps}
            value={value}
            onChange={event => {
                setValue(event.currentTarget.value);
                restProps.onChange(event);
            }}
        />
    );
};

TextArea.propTypes = {
    className: PropTypes.string,
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    disabled: PropTypes.bool,
    id: PropTypes.string,
    onChange: PropTypes.func,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

TextArea.defaultProps = {
    className: '',
    disabled: false,
    onChange: () => { },
    onClick: () => { },
    onFocus: () => { },
    onBlur: () => { }
};