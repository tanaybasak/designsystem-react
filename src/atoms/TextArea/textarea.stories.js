import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
//@update-path-build-start
import TextArea from './TextArea';
//@update-path-build-end

storiesOf('TextArea', module)
  .add(
    'default',
    () => (
      <TextArea
        aria-label="comments"
        disabled={boolean('Disabled', false)}
        onBlur={action('onBlur')}
        onChange={action('onChange')}
        onClick={action('onClick')}
        onFocus={action('onFocus')}
        placeholder={text('Placeholder', 'Placeholder Text')}
      />
    ),
    {
      info: {
        text: `Description About TextArea Component \n

      import { TextArea } from '@patron/patron-react/textarea';`
      }
    }
  )
  .add(
    'with error',
    () => (
      <div className="hcl-form-group">
        <TextArea
          aria-label="comments"
          data-invalid="true"
          disabled={boolean('Disabled', false)}
          onBlur={action('onBlur')}
          onChange={action('onChange')}
          onClick={action('onClick')}
          onFocus={action('onFocus')}
          placeholder={text('Placeholder', 'Placeholder Text')}
        />
        <label className="hcl-label" htmlFor="labeltext">
          Label text
        </label>
        <div className="hcl-error-msg">Validation message here</div>
      </div>
    ),
    {
      info: {
        text: `Description About TextArea Component \n
  
        import { TextArea } from '@patron/patron-react/textarea';`
      }
    }
  )
  .add(
    'with helpertext',
    () => (
      <div className="hcl-form-group">
        <TextArea
          aria-label="comments"
          disabled={boolean('Disabled', false)}
          onBlur={action('onBlur')}
          onChange={action('onChange')}
          onClick={action('onClick')}
          onFocus={action('onFocus')}
          placeholder={text('Placeholder', 'Placeholder Text')}
        />
        <label className="hcl-label" htmlFor="labeltext">
          Label text
        </label>
        <div className="hcl-helper-text">Optional Helper text goes here</div>
      </div>
    ),
    {
      info: {
        text: `Description About TextArea Component \n
  
        import { TextArea } from '@patron/patron-react/textarea';`
      }
    }
  );
