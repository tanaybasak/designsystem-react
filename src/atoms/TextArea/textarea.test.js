import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextArea from './TextArea';

describe('<TextArea> component', () => {
  it('snapshots/renders TextArea correctly', () => {
    const tree = renderer.create(<TextArea />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulate change event in TextArea and is called Once', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<TextArea type="text" onChange={mockCallBack} />);
    wrapper
      .find('textarea')
      .simulate('change', { currentTarget: { value: 'Hello' } });
    expect(mockCallBack.mock.calls.length).toEqual(1);
    expect(wrapper.find('textarea').props().value).toBe('Hello');
  });

  it('simulate click, onfocus event in TextArea and is called Once', () => {
    const mockCallBackclick = jest.fn().mockReturnValue(undefined);
    const mockCallBackfocus = jest.fn().mockReturnValue(undefined);
    const mockCallBackchange = jest.fn().mockReturnValue(undefined);
    const mockCallBackblur = jest.fn().mockReturnValue(undefined);
    const wrapper = shallow(<TextArea type="text" />);
    wrapper.find('textarea').simulate('click', { target: {} });
    wrapper.find('textarea').simulate('focus', { target: {} });
    wrapper
      .find('textarea')
      .simulate('change', { currentTarget: { value: 'Hello' } });
    wrapper.find('textarea').simulate('blur', { target: {} });
    expect(mockCallBackclick()).toEqual(undefined);
    expect(mockCallBackfocus()).toEqual(undefined);
    expect(mockCallBackchange()).toEqual(undefined);
    expect(mockCallBackblur()).toEqual(undefined);
  });
});
