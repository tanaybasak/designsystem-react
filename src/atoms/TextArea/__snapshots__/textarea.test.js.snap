// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea> component snapshots/renders TextArea correctly 1`] = `
<textarea
  className="hcl-textarea"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  value=""
/>
`;
